src/FlightDisplay/PreFlightCheckList.qml

```diff
@@ -42,9 +42,14 @@ Rectangle {
                 group = checkListRepeater.itemAt(index + 1)
                 group.enabled = true
                 group._checked = true
+            } else if ((index + 1) == checkListRepeater.count) {
```

Comment:
This won't work unless the group order is enforced. And you added the feature to allow order to not be enforced. Is that right?

Reply:
The fix at quick glance is to walk the list using checkListRepeater.itemAt(...).passed for testing

Reply:
Yup you're right, I've fixed the logic.